packages/xchain-ethereum/src/utils.ts

```diff
@@ -92,7 +92,12 @@ export const validateAddress = (address: Address): boolean => {
  */
 export const getTokenAddress = (asset: Asset): string | null => {
   try {
-    return ethers.utils.getAddress(asset.symbol.slice(asset.ticker.length + 1).substr(2))
+    const assetAddress = asset.symbol.slice(asset.ticker.length + 1)
+
+    if (assetAddress.startsWith('0x') || assetAddress.startsWith('0X'))
+      return ethers.utils.getAddress(assetAddress.substr(2))
```

Comment:
// strip 0X only - 0x is still valid
const strip0X = assetAddress.replace(/\^0X/, '')
ethers.utils.getAddress(strip0X)

Reply:
Since this `0X` it' s a THORChain/Midgard related "issue", we should handle it in ASGDX, but not in `xchain-eth`. `strip0X` might be a helper in `asgardex-util` then.